docs/release.sh

```diff
@@ -134,33 +105,65 @@ invalidate_cache() {
 		echo "\"$dst$file\"$comma" >> batchfile
 	done
 
-	echo "]}, \"CallerReference\":" >> batchfile
-	echo "\"$(date)\"}'" >> batchfile
+	echo "]}, \"CallerReference\":\"$(date)\"}'" >> batchfile
 
-
-	echo "-----"
-	cat batchfile
-	echo "-----"
 	sh batchfile
-	echo "-----"
 }
 
+main() {
+	[ "$AWS_S3_BUCKET" ] || usage
+
+	# Make sure there is an awsconfig file
+	export AWS_CONFIG_FILE=$(pwd)/awsconfig
+	[ -e "$AWS_CONFIG_FILE" ] || usage
 
-if [ "$OPTIONS" != "--dryrun" ]; then
-	setup_s3
-fi
+	# Get the version
+	VERSION=$(cat VERSION)
 
-# Default to only building the version specific docs so we don't clober the latest by accident with old versions
-if [ "$BUILD_ROOT" == "yes" ]; then
-	echo "Building root documentation"
+	# Disallow pushing dev docs to master
+	if [ "$AWS_S3_BUCKET" == "docs.docker.com" ] && [ "${VERSION%-dev}" != "$VERSION" ]; then
+		echo "Please do not push '-dev' documentation to docs.docker.com ($VERSION)"
+		exit 1
+	fi
+
+	# Clean version - 1.0.2-dev -> 1.0
+	export MAJOR_MINOR="v${VERSION%.*}"
+
+	export BUCKET=$AWS_S3_BUCKET
+	export AWS_DEFAULT_PROFILE=$BUCKET
+
+	# debug variables
+	echo "bucket: $BUCKET, full version: $VERSION, major-minor: $MAJOR_MINOR"
+	echo "cfg file: $AWS_CONFIG_FILE ; profile: $AWS_DEFAULT_PROFILE"
+
+	# create the robots.txt
+	create_robots_txt
+
+	if [ "$OPTIONS" != "--dryrun" ]; then
+		setup_s3
+	fi
+
+	# Default to only building the version specific docs
+	# so we don't clober the latest by accident with old versions
+	if [ "$BUILD_ROOT" == "yes" ]; then
+		echo "Building root documentation"
+		build_current_documentation
+
+		echo "Uploading root documentation"
+		upload_current_documentation
+		[ "$NOCACHE" ] || invalidate_cache
+	fi
+
+	#build again with /v1.0/ prefix
+	sed -i "s/^site_url:.*/site_url: \/$MAJOR_MINOR\//" mkdocs.yml
+	echo "Building the /$MAJOR_MINOR/ documentation"
 	build_current_documentation
-	upload_current_documentation
-	[ "$NOCACHE" ] || invalidate_cache
-fi
-
-#build again with /v1.0/ prefix
-sed -i "s/^site_url:.*/site_url: \/$MAJOR_MINOR\//" mkdocs.yml
-echo "Building the /$MAJOR_MINOR/ documentation"
-build_current_documentation
-upload_current_documentation "/$MAJOR_MINOR/"
-[ "$NOCACHE" ] || invalidate_cache "/$MAJOR_MINOR"
+
+	echo "Uploading the documentation"
+	upload_current_documentation "/$MAJOR_MINOR/"
+
+	# Invalidating cache
+	[ "$NOCACHE" ] || invalidate_cache "/$MAJOR_MINOR"
+}
+
+main
```

Comment:
@jfrazelle we could either add a `| sed 's/ /%20/g'` or urlencode each line in the for loop:
```
echo "\"$dst$file\"$comma" | python -c 'import urllib; import sys; print urllib.quote(sys.stdin.readline()[:-1])' >> batchfile
```